software/teachee-desktop/src/sample_source/ft.rs

```diff
@@ -0,0 +1,50 @@
+use std::{thread, time::Duration};
+
+use libftd2xx::{BitMode, Ft232h, FtdiCommon};
+
+use super::{Channel, Result, SampleSource};
+
+const MASK: u8 = 0xFF;
+const LATENCY_TIMER: Duration = Duration::from_millis(16);
+const IN_TRANSFER_SIZE: u32 = 0x10000;
+const RX_BUF_SIZE: usize = 100_000;
```

Comment:
Since you are calling `read_all()` and emptying out the FTDI chip entirely with each read. I wonder if this buffer size should match the size of the FTDI's FIFO. 1KB rather than 100KB?